graph/push.go

```diff
@@ -442,29 +443,22 @@ func (s *TagStore) pushV2Image(r *registry.Session, img *image.Image, endpoint *
 		os.Remove(tf.Name())
 	}()
 
-	ts, err := tarsum.NewTarSum(arch, true, tarsum.Version1)
+	h := sha256.New()
+	size, err := bufferToFile(tf, io.TeeReader(arch, h))
```

Comment:
right - not necessary for this PR.

Once this is merged, I'll submit 2 separate PRs for caching checksums on pull and gzip of content on push.